### PR TITLE
Move IMeterFactory to System.Diagnostics.DiagnosticSource

### PR DESCRIPTION
**Pull request by tarekgh**
The changes involve moving the interfaces from the `Microsoft.Extensions.Diagnostics.Abstraction` library to the `System.Diagnostics.DiagnosticSource` library. This enables other System libraries to utilize these interfaces. While these libraries can depend on other System libraries, they cannot depend directly on Microsoft.Extensions libraries due to layering constraints.

Please be aware that the `Microsoft.Extensions.Diagnostics.Abstraction` library is being phased out, as all interfaces have been relocated from this library.

https://github.com/dotnet/runtime/issues/86961#issuecomment-1613616354

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @noahfalk @ericstj @stephentoub 

@eerhardt @JamesNK we'll need to remove any reference for `Microsoft.Extensions.Diagnostics.Abstraction` from aspnet core builds.

**Comment by stephentoub** [comment]
> Please be aware that the Microsoft.Extensions.Diagnostics.Abstraction library is being phased out, as all interfaces have been relocated from this library.

I thought @noahfalk mentioned something about working on a change that would be adding many more types to this assembly?

**Comment by tarekgh** [comment]
> I thought @noahfalk Noah Falk FTE mentioned something about working on a change that would be adding many more types to this assembly?

We'll bring it back when we need it again. for now, this library is empty.

**Comment by stephentoub** [comment]
> We'll bring it back when we need it again.

Do we know appx when that will be? Are we talking next week? Next release?

File: src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MeterFactoryExtensions.cs [hunk 1]
```diff
@@ -3,9 +3,8 @@
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics.Metrics;
 
-namespace Microsoft.Extensions.Diagnostics.Metrics
+namespace System.Diagnostics.Metrics
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MeterFactoryExtensions.cs:7]: Do we have tests for MeterFactoryExtensions.Create somewhere?

**Review by stephentoub — APPROVED** [review]
src changes LGTM

**Comment by noahfalk** [comment]
> Do we know appx when that will be? Are we talking next week? Next release?

This release, hopefully but not necessarily before July 17th. Fowler tells me he had gotten it approved to bring in late features as part of R9 work and this was one of them. I had been meaning to talk to you Stephen because I wasn't sure how broadly Fowler had communicated around it.

**Review by antonfirsov — APPROVED** [review]

File: src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MeterFactoryExtensions.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MeterFactoryExtensions.cs:7]: Yes it is in https://github.com/dotnet/runtime/blob/f9f7f8079377039a8d42269434a41ca55d14d591/src/libraries/Microsoft.Extensions.Diagnostics/tests/MetricsTests.cs#L43

**Review by noahfalk — APPROVED** [review]
👍

**Pull request merged**